gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -5050,12 +5071,15 @@ gnc_plugin_page_register_cmd_scrub_all (GtkAction* action,
             g_free (progress_msg);
         }
 
+        if (abort_scrub) break;
+
         scrub_split (split);
 
         PINFO ("Finished processing split %d of %d",
                curr_split_no + 1, split_count);
     }
 
+    g_signal_handler_disconnect (G_OBJECT (window), scrub_kp_handler_ID);
```

Comment:
I don't think you want to disconnect the handler, that will would mean it works only the first time check & repair is run. You *do* need to `abort_scrub = FALSE;` here other wise subsequent runs of check & repair will do only one split.